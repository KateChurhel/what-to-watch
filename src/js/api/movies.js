/* istanbul ignore file */
import { sendGetRequest } from './index';

export const getCategoryData = (category, data) => sendGetRequest(`/${category}/popular`, data);

export const getDetailsData = (category, id, data) => sendGetRequest(`/${category}/${id}`, data);

export const getSearchResult = (category, data) => sendGetRequest(`/search/${category}`, data);

export const getDiscoverResult = (data) => sendGetRequest('/discover/movie', data);

export const getGenres = () => sendGetRequest('/genre/movie/list');
