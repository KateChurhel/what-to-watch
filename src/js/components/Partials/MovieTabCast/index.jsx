// libraries
import React from 'react';
import PropTypes from 'prop-types';
// constants
import { URL_IMAGE } from '../../../constants/url';

const MovieTabCast = ({ cast = [] }) => (
  <div className="movie-card__tab-content-wrapper">
    {cast.length ? (
      <ol className="movie-card__casts">
        {cast.map(({
          cast_id: id, character, name: profileName, profile_path: profilePath, gender,
        }) => {
          const reservedImage = gender === 2 ? '/img/woman-profile.jpeg' : '/img/man-profile.jpeg';

          return (
            <li key={`cast-${id}`} className="movie-card__cast-wrapper">
              <img
                alt={profileName}
                className="movie-card__cast-image"
                height="175"
                loading="lazy"
                src={profilePath ? `${URL_IMAGE.castPhoto}${profilePath}` : reservedImage}
                width="138"
              />
              <p className="movie-card__cast-name">{profileName}</p>
              <p className="movie-card__cast-character">{character}</p>
            </li>
          );
        })}
      </ol>
    ) : (<p>Cast not found</p>)}
  </div>
);

MovieTabCast.defaultProps = {
  cast: [],
};

MovieTabCast.propTypes = {
  cast: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    character: PropTypes.string,
    name: PropTypes.string,
    profile_path: PropTypes.string,
    gender: PropTypes.number,
  })),
};

export default MovieTabCast;
