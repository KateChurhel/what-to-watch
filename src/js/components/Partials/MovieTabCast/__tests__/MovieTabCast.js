import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { MOCK_MOVIES } from '../../../../mocks/movies';
import MovieTabCast from '../index';

describe('Movies Tab Cast: ', () => {
  it('correctly renders with data', () => {
    const tree = shallow(<MovieTabCast cast={MOCK_MOVIES[0].credits.cast} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('correctly renders without data', () => {
    const tree = shallow(<MovieTabCast />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
