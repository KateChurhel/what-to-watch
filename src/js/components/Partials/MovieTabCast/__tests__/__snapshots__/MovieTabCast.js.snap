// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies Tab Cast:  correctly renders with data 1`] = `
<div
  className="movie-card__tab-content-wrapper"
>
  <ol
    className="movie-card__casts"
  >
    <li
      className="movie-card__cast-wrapper"
      key="cast-26"
    >
      <img
        alt="Ryan Gosling"
        className="movie-card__cast-image"
        height="175"
        loading="lazy"
        src="https://image.tmdb.org/t/p/w500/lyUyVARQKhGxaxy0FbPJCQRpiaW.jpg"
        width="138"
      />
      <p
        className="movie-card__cast-name"
      >
        Ryan Gosling
      </p>
      <p
        className="movie-card__cast-character"
      >
        Sebastian Wilder
      </p>
    </li>
    <li
      className="movie-card__cast-wrapper"
      key="cast-27"
    >
      <img
        alt="Emma Stone"
        className="movie-card__cast-image"
        height="175"
        loading="lazy"
        src="https://image.tmdb.org/t/p/w500/cZ8a3QvAnj2cgcgVL6g4XaqPzpL.jpg"
        width="138"
      />
      <p
        className="movie-card__cast-name"
      >
        Emma Stone
      </p>
      <p
        className="movie-card__cast-character"
      >
        Mia Dolan
      </p>
    </li>
    <li
      className="movie-card__cast-wrapper"
      key="cast-34"
    >
      <img
        alt="John Legend"
        className="movie-card__cast-image"
        height="175"
        loading="lazy"
        src="/img/woman-profile.jpeg"
        width="138"
      />
      <p
        className="movie-card__cast-name"
      >
        John Legend
      </p>
      <p
        className="movie-card__cast-character"
      >
        Keith
      </p>
    </li>
    <li
      className="movie-card__cast-wrapper"
      key="cast-32"
    >
      <img
        alt="Rosemarie DeWitt"
        className="movie-card__cast-image"
        height="175"
        loading="lazy"
        src="/img/man-profile.jpeg"
        width="138"
      />
      <p
        className="movie-card__cast-name"
      >
        Rosemarie DeWitt
      </p>
      <p
        className="movie-card__cast-character"
      >
        Laura Wilder
      </p>
    </li>
  </ol>
</div>
`;

exports[`Movies Tab Cast:  correctly renders without data 1`] = `
<div
  className="movie-card__tab-content-wrapper"
>
  <p>
    Cast not found
  </p>
</div>
`;
